modin/pandas/test/dataframe/test_reduce.py

```diff
@@ -306,6 +306,22 @@ def test_sum(data, axis, skipna, is_transposed):
     df_equals(modin_result, pandas_result)
 
 
+@pytest.mark.parametrize("nptypes", [["int64"]])
+@pytest.mark.parametrize("pdtypes", [["Int64"]])
+def test_dtype_consistency(nptypes, pdtypes):
+    # test for issue #6781
+
+    # Test the resultant dtypes with numpy datatypes
+    for nptype in nptypes:
+        res_dtype = pd.DataFrame([1, 2, 3, 4], dtype=nptype).sum().dtype
+        assert res_dtype == np.dtype(nptype)
+
+    # Test the resultant dtypes with pandas datatypes
+    for pdtype in pdtypes:
+        res_dtype = pd.DataFrame([1, 2, 3, 4], dtype=pdtype).sum().dtype
+        assert res_dtype == pandas.api.types.pandas_dtype(pdtype)
```

Comment:
Let's use one logic for all dtypes as it done in `dataframe.py`.

```suggestion
@pytest.mark.parametrize("dtype", ["int64", "Int64"])
def test_dtype_consistency(dtype):
    # test for issue #6781
    res_dtype = pd.DataFrame([1, 2, 3, 4], dtype=dtype).sum().dtype
    assert res_dtype == pandas.api.types.pandas_dtype(pdtype)
```

Also, your test fails for the HDK engine, but for now it's ok, let's just skip this case. For this you can use the following code `@pytest.mark.skipif(Engine.get() == "Native", reason="Fails on HDK")`.

**Note:** `Engine` can be imported from `modin.config`.

Reply:
Thanks for the suggestion. I have modified the code